### PR TITLE
pd: update online recovery related fields

### PR DESCRIPTION
**Pull request by Connor1996**
Add more fields related with online recovery

### CODE REVIEW

**Comment by Connor1996** [comment]
PTAL @v01dstar

File: proto/pdpb.proto [hunk 1]
```diff
@@ -598,23 +601,48 @@ message StoreHeartbeatRequest {
...
-    // Detailed store report that is only filled up on PD's demand for online unsafe recover.
+    // Detailed store report that is only filled up on PD's demand for online unsafe recovery.
     StoreReport store_report = 3;
     replication_modepb.StoreDRAutoSyncStatus dr_autosync_status = 4;
 }
 
+message RemovePeers {
+    uint64 region_id = 1;
+    repeated metapb.Peer peers = 2;
+}
```
Comment (v01dstar) [inline proto/pdpb.proto:612]:
```suggestion
message DemoteFailedVoters {
    uint64 region_id = 1;
    repeated metapb.Peer failed_voters = 2;
}
```

File: proto/pdpb.proto [hunk 2]
```diff
@@ -598,23 +601,48 @@ message StoreHeartbeatRequest {
...
+    // Create empty regions to fill the key range hole.
     repeated metapb.Region creates = 1;
-    repeated metapb.Region updates = 2;
-    repeated uint64 deletes = 3;
+    // Update the meta of the regions, including peer lists, epoch and key range.
+    repeated metapb.Region updates = 2 [deprecated=true];
+    // Tombstone the peers on the store locally.
+    repeated uint64 tombstones = 3;
+    // Issue remove-peer conf change on the regions.
+    repeated RemovePeers removes = 4;
```
Comment (v01dstar) [inline proto/pdpb.proto:629]:
```suggestion
    repeated DemoteFailedVoters demotes = 4;
```

File: proto/pdpb.proto [hunk 3]
```diff
@@ -598,23 +601,48 @@ message StoreHeartbeatRequest {
...
+    // Create empty regions to fill the key range hole.
     repeated metapb.Region creates = 1;
-    repeated metapb.Region updates = 2;
-    repeated uint64 deletes = 3;
+    // Update the meta of the regions, including peer lists, epoch and key range.
+    repeated metapb.Region updates = 2 [deprecated=true];
+    // Tombstone the peers on the store locally.
+    repeated uint64 tombstones = 3;
+    // Issue conf change that demote voters on failed stores to learners on the regions.
+    repeated DemoteFailedVoters demotes = 4;
```
[the remark this answers is not in the document]
Comment (Connor1996) [reply proto/pdpb.proto:629]: It's an individual code path from PD normal scheduler. We need to dispatch all the ops and then wait apply to report.

**Review by BusyJay — APPROVED** [review]

File: proto/pdpb.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (v01dstar) [reply proto/pdpb.proto:629]: In phase 1 implementation, without force leader, we have to specialize these kinds of operation. With force leader, in theory, we can indeed utilize existing PD functionalities to achieve the same goal. It is just a choice we made regarding the timeline and complexity.

**Review by v01dstar — APPROVED** [review]

File: proto/pdpb.proto [hunk 4]
```diff
@@ -588,6 +588,9 @@ message StoreStats {
 message PeerReport {
     raft_serverpb.RaftLocalState raft_state = 1;
     raft_serverpb.RegionLocalState region_state = 2;
+    bool is_force_leader = 3;
```
Comment (v01dstar) [inline proto/pdpb.proto:591]: Typo? is_forced_leader?

Comment (Connor1996) [reply proto/pdpb.proto:591]: no, I'd like use `force_leader` to keep consistency with TiKV

**Comment by Connor1996** [comment]
Will merge it until the last PR of TiKV and PD is ready to be merged after review

**Pull request merged**